pkg/client/dtclient/config_client.go

```diff
@@ -897,3 +902,33 @@ func validateSloV1Payload(payload []byte) error {
 	}
 	return nil
 }
+
+// ValidateDashboardPayload returns an error if the JSON data is invalid or if the payload is not a V1 payload
+// expectV1: if true it's expected to be a V1 dashboard, else V2
+// Classic dashboard: "tiles" is an array
+// Platform dashboard: "tiles" is an object
+func ValidateDashboardPayload(payload []byte, expectV1 bool) error {
+	type DashboardV2Keys struct {
+		Tiles any `json:"tiles"`
+	}
+	parsedPayload := DashboardV2Keys{}
+	err := json.Unmarshal(payload, &parsedPayload)
+	if err != nil {
+		return err
+	}
+
+	if expectV1 {
+		// map should only be used in V2
+		if _, isMap := parsedPayload.Tiles.(map[string]any); isMap {
+			return errors.New("tried to deploy a classic dashboard configuration to platform dashboard")
+		}
+		return nil
+	}
+
+	// array should only be used in V1
+	if _, isArray := parsedPayload.Tiles.([]any); isArray {
+		return errors.New("tried to deploy a platform dashboard configuration to classic dashboard")
```

Comment:
Just wondering: do we use the term "platform dashboard" anywhere else? Would that mean anything to a user?

Reply:
I adjusted it a bit, to not mention "platform"